source/Nevermore.IntegrationTests/QueryableIntegrationFixture.cs

```diff
@@ -1475,5 +1475,118 @@ public async Task AsyncEnumeration()
+            };
+
+            foreach (var c in testCustomers)
+            {
+                t.Insert(c);
+            }
+
+            await t.CommitAsync();
+
+            var customers = t.Queryable<Customer>().Select(c => c.FirstName);
+
+            customers.Should().BeEquivalentTo("Alice", "Bob", "Charlie");
+        }
+
+        [Test]
+        public async Task ProjectJsonField()
+        {
+            using var t = Store.BeginTransaction();
+
+            var testCustomers = new[]
+            {
+                new Customer { FirstName = "Alice", LastName = "Apple", IsEmployee = true },
+                new Customer { FirstName = "Bob", LastName = "Banana", IsEmployee = false },
+                new Customer { FirstName = "Charlie", LastName = "Cherry", IsEmployee = true }
+            };
+
+            foreach (var c in testCustomers)
+            {
+                t.Insert(c);
+            }
+
+            await t.CommitAsync();
+
+            var customers = t.Queryable<Customer>().Select(c => new { c.FirstName, c.IsEmployee });
+
+            customers.Should().BeEquivalentTo(new[] {
+                new { FirstName = "Alice", IsEmployee = true },
+                new { FirstName = "Bob", IsEmployee = false },
+                new { FirstName = "Charlie", IsEmployee = true }
+            });
+        }
+
+        [Test]
+        public async Task ProjectValueType()
+        {
+            using var t = Store.BeginTransaction();
+
+            var testCustomers = new[]
+            {
+                new Customer { FirstName = "Alice", LastName = "Apple", IsEmployee = true },
+                new Customer { FirstName = "Bob", LastName = "Banana", IsEmployee = false },
+                new Customer { FirstName = "Charlie", LastName = "Cherry", IsEmployee = true }
+            };
+
+            foreach (var c in testCustomers)
+            {
+                t.Insert(c);
+            }
+
+            await t.CommitAsync();
+
+            var customers = await t.Queryable<Customer>().Select(c => c.IsEmployee).ToListAsync();
+
+            customers.Should().BeEquivalentTo(new[] { true, false, true });
+        }
+
+        [Test]
+        public async Task ProjectTypeWithParameterizedConstructor()
+        {
+            using var t = Store.BeginTransaction();
+
+            var testCustomers = new[]
+            {
+                new Customer { FirstName = "Alice", LastName = "Apple", IsEmployee = true },
+                new Customer { FirstName = "Bob", LastName = "Banana", IsEmployee = false },
+                new Customer { FirstName = "Charlie", LastName = "Cherry", IsEmployee = true }
+            };
+
+            await t.InsertManyAsync(testCustomers);
+            await t.CommitAsync();
+
+            var customers = await t.Queryable<Customer>().Select(c => new CustomerProjection(c.FirstName, c.IsEmployee)).ToListAsync();
+
+            customers.Should().BeEquivalentTo(new[]
```

Comment:
BeEquivalentTo doesn't care about row ordering. Is that significant here? (you can add it by setting options on the BeEquivalentTo)

Reply:
I don't think it really matters, since what this test cares about is the successful projection. Still, if a more restrictive assertion like `Equal()` can be used it should be.

Reply:
Actually fair enough to use the unordered check here. SQL server doesn't guarantee row order unless you have an OrderBy clause, and you don't have one here. Adding that in the mix would be a different test.